File: Searchings/BinarySearch/BinarySearch.py
def binarySearch(list,low,high,n):
    mid1=(low+high)//2
    for i in range(len(list)):
        mid=list[mid1]
    if n==mid:
        return mid
    else:
        if n<mid:
            return binarySearch(list,low,mid,n)
        else:
            return binarySearch(list,mid+1,high-1,n)
list=list(map(int,input().split()))
high=len(list)
list.sort()
low=0
n=int(input())
print(binarySearch(list,low,high,n))

File: Searchings/BinarySearch/BinarySearch.cpp
#include <bits/stdc++.h>
using namespace std;

void binarySearch(int arr[],int n,int high,int low){
    int mid = (high+low)/2;
    if(arr[mid] == n){
        cout<<"Element found at index "<<mid<<endl;
        return;
    }
    else if(arr[mid] > n){
        binarySearch(arr,n,mid-1,low);
    }
    else{
        binarySearch(arr,n,high,mid+1);
    }
}

int main(){
    int array[]={1,2,3,4,5,6,7,8,9,10};
    cout<<"Enter the number to be searched:";
    int n;
    cin>>n;
    binarySearch(array,n,9,4);
}